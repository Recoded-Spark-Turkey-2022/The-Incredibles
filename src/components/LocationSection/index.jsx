import React, { useState } from 'react';
import Worldmap from '../../assets/pics/homepage/world-map.svg';
import LocationSlide from '../LocationSlides';
import { useTranslation } from 'react-i18next';
import Slider from "react-slick";
import '../../style/slick.css';
import '../../style/slick-theme.css';

function Location() {
  const [t, i18n] = useTranslation();

  const slides = [
    {
      text: `${t('home.location.text1')}`,
      author: 'Omer,',
    },
    {
      text: `${t('home.location.text2')}`,
      author: 'Ahmad,',
    },
    {
      text: `${t('home.location.text3')}`,
      author: 'Miray,',
    },
    {
      text: `${t('home.location.text4')}`,
      author: 'Rabia,',
    },
  ];
  const settings = {
    dots: true,
    infinite: true,
    speed: 1000,
    slidesToShow: 1,
    slidesToScroll: 1,
    autoplay: true,
    cssEase: "linear",
    autoplaySpeed: 5000,
    swipeToSlide: true,

  };




  return (
    <div
      name="main"
      className=" flex justify-between max-lg:flex-col p-5 pt-20 gap-2 max-w-screen-xl m-auto"
    >
      <div name="Photo" className=" h-fit m-auto">
        <img src={Worldmap} alt="world-map" />
      </div>

      <div
        name="paragraphs"
        className=" h-fit lg:w-1/2 flex justify-center flex-col align-middle max-lg:mt-5"
      >
       

        <div className=" ">
        <Slider {...settings}>
          {slides.map((slide, slideIndex) => (
            <LocationSlide key={slideIndex} data={slide} />
           
          ))}
         

          </Slider>
        </div>
      </div>
    </div>
  );
}

export default Location;
