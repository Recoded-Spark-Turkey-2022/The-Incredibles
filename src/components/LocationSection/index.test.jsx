import renderer from 'react-test-renderer';
import LocationSection from '.';

describe('LocationSection', () =>
  it('renders correctly', () => {
    const domTree = renderer.create(<LocationSection />).toJSON();
    expect(domTree).toMatchSnapshot();
  }));
