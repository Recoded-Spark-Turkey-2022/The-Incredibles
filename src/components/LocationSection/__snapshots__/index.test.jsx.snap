// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationSection renders correctly 1`] = `
<div
  className="container max-w-screen-xl m-auto p-10 max-md:px-5"
>
  <div
    className=" flex justify-between max-md:flex-col pt-12 gap-5"
    name="main"
  >
    <div
      className=" h-fit m-auto"
      name="Photo"
    >
      <img
        alt="world-map"
        src="world-map.svg"
      />
    </div>
    <div
      className=" h-fit md:w-1/2 flex justify-center flex-col align-middle max-md:mt-5"
      name="paragraphs"
    >
      <div
        className=" "
      >
        <div
          className="slick-slider slick-initialized"
          dir="ltr"
        >
          <button
            className="slick-arrow slick-prev"
            data-role="none"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
            type="button"
          >
             
            Previous
          </button>
          <div
            className="slick-list"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={null}
            onMouseMove={null}
            onMouseUp={[Function]}
            onTouchCancel={null}
            onTouchEnd={[Function]}
            onTouchMove={null}
            onTouchStart={[Function]}
            style={Object {}}
          >
            <div
              className="slick-track"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "WebkitTransform": "translate3d(0px, 0px, 0px)",
                  "WebkitTransition": "",
                  "msTransform": "translateX(0px)",
                  "opacity": 1,
                  "transform": "translate3d(0px, 0px, 0px)",
                  "transition": "",
                }
              }
            >
              <div
                aria-hidden={true}
                className="slick-slide slick-cloned"
                data-index={-1}
                onClick={[Function]}
                style={
                  Object {
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div>
                    <p
                      className=" max-md:text-lg max-md:text-center sm:text-xl text-base"
                    >
                      home.location.text4
                    </p>
                    <p
                      className="pt-8 max-md:text-center"
                    >
                      Rabia,
                      <br />
                      home.location.team
                    </p>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={false}
                className="slick-slide slick-active slick-current"
                data-index={0}
                onClick={[Function]}
                style={
                  Object {
                    "outline": "none",
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div>
                    <p
                      className=" max-md:text-lg max-md:text-center sm:text-xl text-base"
                    >
                      home.location.text1
                    </p>
                    <p
                      className="pt-8 max-md:text-center"
                    >
                      Omer,
                      <br />
                      home.location.team
                    </p>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={true}
                className="slick-slide"
                data-index={1}
                onClick={[Function]}
                style={
                  Object {
                    "outline": "none",
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div>
                    <p
                      className=" max-md:text-lg max-md:text-center sm:text-xl text-base"
                    >
                      home.location.text2
                    </p>
                    <p
                      className="pt-8 max-md:text-center"
                    >
                      Ahmad,
                      <br />
                      home.location.team
                    </p>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={true}
                className="slick-slide"
                data-index={2}
                onClick={[Function]}
                style={
                  Object {
                    "outline": "none",
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div>
                    <p
                      className=" max-md:text-lg max-md:text-center sm:text-xl text-base"
                    >
                      home.location.text3
                    </p>
                    <p
                      className="pt-8 max-md:text-center"
                    >
                      Miray,
                      <br />
                      home.location.team
                    </p>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={true}
                className="slick-slide"
                data-index={3}
                onClick={[Function]}
                style={
                  Object {
                    "outline": "none",
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div>
                    <p
                      className=" max-md:text-lg max-md:text-center sm:text-xl text-base"
                    >
                      home.location.text4
                    </p>
                    <p
                      className="pt-8 max-md:text-center"
                    >
                      Rabia,
                      <br />
                      home.location.team
                    </p>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={true}
                className="slick-slide slick-cloned"
                data-index={4}
                onClick={[Function]}
                style={
                  Object {
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div>
                    <p
                      className=" max-md:text-lg max-md:text-center sm:text-xl text-base"
                    >
                      home.location.text1
                    </p>
                    <p
                      className="pt-8 max-md:text-center"
                    >
                      Omer,
                      <br />
                      home.location.team
                    </p>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={true}
                className="slick-slide slick-cloned"
                data-index={5}
                onClick={[Function]}
                style={
                  Object {
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div>
                    <p
                      className=" max-md:text-lg max-md:text-center sm:text-xl text-base"
                    >
                      home.location.text2
                    </p>
                    <p
                      className="pt-8 max-md:text-center"
                    >
                      Ahmad,
                      <br />
                      home.location.team
                    </p>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={true}
                className="slick-slide slick-cloned"
                data-index={6}
                onClick={[Function]}
                style={
                  Object {
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div>
                    <p
                      className=" max-md:text-lg max-md:text-center sm:text-xl text-base"
                    >
                      home.location.text3
                    </p>
                    <p
                      className="pt-8 max-md:text-center"
                    >
                      Miray,
                      <br />
                      home.location.team
                    </p>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={true}
                className="slick-slide slick-cloned"
                data-index={7}
                onClick={[Function]}
                style={
                  Object {
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div>
                    <p
                      className=" max-md:text-lg max-md:text-center sm:text-xl text-base"
                    >
                      home.location.text4
                    </p>
                    <p
                      className="pt-8 max-md:text-center"
                    >
                      Rabia,
                      <br />
                      home.location.team
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <button
            className="slick-arrow slick-next"
            data-role="none"
            onClick={[Function]}
            style={
              Object {
                "display": "block",
              }
            }
            type="button"
          >
             
            Next
          </button>
          <ul
            className="slick-dots"
            onMouseEnter={null}
            onMouseLeave={null}
            onMouseOver={null}
            style={
              Object {
                "display": "block",
              }
            }
          >
            <li
              className="slick-active"
            >
              <button
                onClick={[Function]}
              >
                1
              </button>
            </li>
            <li
              className=""
            >
              <button
                onClick={[Function]}
              >
                2
              </button>
            </li>
            <li
              className=""
            >
              <button
                onClick={[Function]}
              >
                3
              </button>
            </li>
            <li
              className=""
            >
              <button
                onClick={[Function]}
              >
                4
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
